File: packages/create-meowkit/templates/claude/skills/meow:multimodal/scripts/tests/run_tests.py
#!/usr/bin/env python3
"""
Test runner for meow:multimodal scripts.

Runs all test_*.py files in this directory.
No external test framework required — uses built-in assertions.
"""

import subprocess
import sys
from pathlib import Path


def main():
    test_dir = Path(__file__).parent
    test_files = sorted(test_dir.glob("test_*.py"))

    if not test_files:
        print("No test files found!")
        sys.exit(1)

    print(f"Running {len(test_files)} test files for meow:multimodal...\n")

    total_passed = 0
    total_failed = 0
    failed_files = []

    for test_file in test_files:
        print(f"{'=' * 60}")
        print(f"Running: {test_file.name}")
        print('=' * 60)

        result = subprocess.run(
            [sys.executable, str(test_file)],
            capture_output=False,
        )

        if result.returncode != 0:
            failed_files.append(test_file.name)

    print(f"\n{'=' * 60}")
    print("All Tests Summary")
    print('=' * 60)
    print(f"Total test files: {len(test_files)}")
    print(f"Passed: {len(test_files) - len(failed_files)}")
    print(f"Failed: {len(failed_files)}")

    if failed_files:
        print("\nFailed tests:")
        for f in failed_files:
            print(f"  - {f}")
        sys.exit(1)
    else:
        print("\n✓ All tests passed!")
        sys.exit(0)


if __name__ == '__main__':
    main()

File: .claude/skills/meow:docs-finder/scripts/tests/run-tests.js
#!/usr/bin/env node

/**
 * Test runner for meow:docs-finder scripts
 */

const { spawn } = require('child_process');
const path = require('path');

const tests = [
  'test-detect-source.js',
  'test-fetch-context7.js',
  'test-fetch-chub.js',
  'test-analyze-results.js',
];

function runTest(testFile) {
  return new Promise((resolve, reject) => {
    console.log(`\n${'='.repeat(60)}`);
    console.log(`Running: ${testFile}`);
    console.log('='.repeat(60));

    const testPath = path.join(__dirname, testFile);
    const proc = spawn('node', [testPath], { stdio: 'inherit' });

    proc.on('close', (code) => {
      if (code === 0) resolve();
      else reject(new Error(`Test failed: ${testFile}`));
    });

    proc.on('error', reject);
  });
}

async function runAllTests() {
  console.log('Running all meow:docs-finder tests...');

  let failedTests = [];

  for (const test of tests) {
    try {
      await runTest(test);
    } catch (error) {
      failedTests.push(test);
    }
  }

  console.log(`\n${'='.repeat(60)}`);
  console.log('All Tests Summary');
  console.log('='.repeat(60));
  console.log(`Total test files: ${tests.length}`);
  console.log(`Passed: ${tests.length - failedTests.length}`);
  console.log(`Failed: ${failedTests.length}`);

  if (failedTests.length > 0) {
    console.log('\nFailed tests:');
    failedTests.forEach((test) => console.log(`  - ${test}`));
    process.exit(1);
  } else {
    console.log('\n✓ All tests passed!');
    process.exit(0);
  }
}

runAllTests();
